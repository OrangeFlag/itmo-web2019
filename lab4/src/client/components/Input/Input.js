import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledInput = styled.input`
  background-color: ___CSS_0___;
  border: 1px solid ___CSS_1___;
  border-radius: 4px;
  padding: 12px 15px;
  margin: 8px 0;
  outline: none;
  font-size: '1rem';
  transition: ___CSS_2___;

  &:focus {
    border: 1px solid ___CSS_3___;
    transition: ___CSS_4___;
  }
`;


class Input extends React.Component {

    state = {
        value: ""
    };

    handleChange = event => {
        this.setState({value: event.target.value});
    };

    render() {
        this.handleChange = this.props.handleChange || this.handleChange;
        this.handleSubmit = this.props.handleSubmit;

        return (
            <StyledInput
                id={this.props.id}
                type={this.props.type}
                name={this.props.name}
                defaultValue={this.state.value}
                onChange={this.handleChange}
                onSubmit={this.handleSubmit}
                required={this.props.required}
                pattern={this.props.pattern}
                placeholder={this.props.placeholder}
            />
        );
    }
}

Input.propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    value: PropTypes.string,
    type: PropTypes.oneOf(['text', 'email', 'password', 'number']),
    required: PropTypes.bool,
    pattern: PropTypes.string,
    placeholder: PropTypes.string,
    handleChange: PropTypes.func
};


export default Input