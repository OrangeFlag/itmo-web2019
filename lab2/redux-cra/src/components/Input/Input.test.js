import React from 'react';
import {mount} from 'enzyme';
import Input from './Input';

describe('Input test', () => {
    it('should render correctly', () => {
        const component = mount(<Input id="new_town" type="text" name="favorite" placeholder="Добавить новый город"
                                       maxLength="100"/>);

        expect(component).toMatchSnapshot();
        component.unmount();
    });

    it('input change should call function', () => {
        const changeFn = jest.fn();
        const component = mount(<Input handleChange={changeFn}/>);

        component.simulate('change', {target: {value: 'Hello'}});

        expect(changeFn).toHaveBeenCalled();
        component.unmount();
    });
});