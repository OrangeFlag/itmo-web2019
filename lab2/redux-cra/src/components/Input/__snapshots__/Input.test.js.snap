// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test should render correctly 1`] = `
<Input
  id="new_town"
  maxLength="100"
  name="favorite"
  placeholder="Добавить новый город"
  type="text"
>
  <styled.input
    defaultValue=""
    id="new_town"
    maxLength="100"
    name="favorite"
    onChange={[Function]}
    placeholder="Добавить новый город"
    type="text"
  >
    <StyledComponent
      defaultValue=""
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "hMNFWJ",
            "rules": Array [
              "
  background-color: ___CSS_0___;
  border: 1px solid ___CSS_1___;
  border-radius: 4px;
  padding: 12px 15px;
  margin: 8px 0;
  outline: none;
  font-size: '1rem';
  transition: ___CSS_2___;

  &:focus {
    border: 1px solid ___CSS_3___;
    transition: ___CSS_4___;
  }
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      id="new_town"
      maxLength="100"
      name="favorite"
      onChange={[Function]}
      placeholder="Добавить новый город"
      type="text"
    >
      <input
        className="sc-bdVaJa hMNFWJ"
        defaultValue=""
        id="new_town"
        maxLength="100"
        name="favorite"
        onChange={[Function]}
        placeholder="Добавить новый город"
        type="text"
      />
    </StyledComponent>
  </styled.input>
</Input>
`;
